components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -331,19 +334,24 @@ open class GleanInternalAPI internal constructor () {
             return false
         }
 
-        var hasPingContent = false
+        var pingSerializationTasks: MutableList<Job> = mutableListOf()
         for (pingName in pingNames) {
-            val hasContent = assembleAndSerializePing(pingName)
-            hasPingContent = hasPingContent || hasContent
+            assembleAndSerializePing(pingName)?.let {
+                pingSerializationTasks.add(it)
+            }
         }
 
         // It should only take a single PingUploadWorker to process all queued pings, so we only
         // want to have one scheduled at a time.
-        if (hasPingContent) {
+        if (pingSerializationTasks.any()) {
+            runBlocking {
```

Comment:
Does this mean that `sendPings` blocks on assembling and serializing pings?  Since that's disk I/O, I'm not sure we can do that.  I think we need a job in which to do this blocking + enqueuing of the worker.

Reply:
Yes, I can rework it to a task that runs in the background too, I was worried about introducing too much async'ery but I think that's a valid point.

Reply:
You are right to worry.  I think we're following the general rule of thumb that I/O is bad and should always be async.  But that could be premature asyncrifying.